thewalrus/quantum/photon_number_distributions.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Functions for calculating the photon number distributions of various states.
+"""
+
+import numpy as np
+from scipy.stats import nbinom
+
+from .conversions import Amat
+from .gaussian_state_tests import is_pure_cov
+
+
+def pure_state_distribution(cov, cutoff=50, hbar=2, padding_factor=2):
+    r""" Calculates the total photon number distribution of a pure state
+    with zero mean.
+
+    Args:
+        cov (array): :math:`2N\times 2N` covariance matrix in xp-ordering
+        cutoff (int): Fock cutoff
+        tol (float): tolerance for determining if displacement is negligible
+        hbar (float): the value of :math:`\hbar` in the commutation
+        padding_factor (int): expanded size of the photon distribution to avoid accumulation of errors
+
+    Returns:
+        (array): Total photon number distribution
+    """
+    if is_pure_cov(cov):
+        A = Amat(cov, hbar=hbar)
+        (n, _) = A.shape
+        N = n // 2
+        B = A[0:N, 0:N]
+        rs = np.arctanh(np.linalg.svd(B, compute_uv=False))
+        return _convolve_squeezed_state_distribution(rs, cutoff=cutoff, padding_factor=padding_factor)[0:cutoff]
+    raise ValueError("The Gaussian state is not pure")
+
+
+def _squeezed_state_distribution(s, cutoff=50, N=1):
```

Comment:
Curious why this is hidden (has it always been hidden?) Sounds like it could be quite useful in some contexts

Reply:
It wasn't actually hidden before. I think that @nquesada mentioned it could be hidden, but I might have misunderstood.

Reply:
It used to be not hidden. The main idea behind this is that most of the functions should take as arguments a `cov` matrix, this one does not.